pkg/kubelet/runonce.go

```diff
@@ -156,3 +156,19 @@ func (kl *Kubelet) isPodRunning(pod *v1.Pod, status *kubecontainer.PodStatus) bo
 	}
 	return true
 }
+
+// getFailContainer returns failed container name for pod.
+func (kl *Kubelet) getFailContainer(pod *v1.Pod) ([]string, error) {
+	status, err := kl.containerRuntime.GetPodStatus(pod.UID, pod.Name, pod.Namespace)
+	containername := make([]string, 0)
+	if err != nil {
+		return containername, fmt.Errorf("Unable to get status for pod %q: %v", format.Pod(pod), err)
+	}
+	for _, c := range pod.Spec.Containers {
```

Comment:
https://github.com/kubernetes/kubernetes/pull/59537#discussion_r168866448

> container status has a name, just iterate over those.

Reply:
@tallclair I think it has iterated in func FindContainerStatusByName below, if I misunderstand it ?

Reply:
What I meant was:
```go
for _, cs := range status.ContainerStatuses {
	if cs.State != kubecontainer.ContainerStateRunning && cs.ExitCode != 0 {
		containerNames = append(containerNames, cs.Name)
	}
}
```

Reply:
Done